### PR TITLE
No more hidden elements

### LINKED ISSUE #66046: Click to go to a hidden element should make it visible
It's pretty common when its parent is not fully viewable.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #66046.

Follow-up of #66082.

r? @kinnison

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -161,17 +161,18 @@ function getSearchElement() {
         return window.history && typeof window.history.pushState === "function";
     }
 
+    function isHidden(elem) {
+        return elem.offsetHeight === 0;
+    }
+
```
Comment (kinnison) [inline src/librustdoc/html/static/main.js]: This function seems odd to have, there's code further down which called an `isHidden()` function you don't appear to have removed, so is this shadowing one?

File: src/librustdoc/html/static/main.js [hunk 2]
```diff
@@ -2615,6 +2652,10 @@ function getSearchElement() {
         insertAfter(popup, getSearchElement());
     }
 
+    handleHashes();
+    highlightSourceLines();
```
Comment (kinnison) [inline src/librustdoc/html/static/main.js]: Would it be more future-proof to just call `onHashChange()` here?

File: src/librustdoc/html/static/main.js (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: You're absolutely right! I removed the old one.

File: src/librustdoc/html/static/main.js (hunk 2 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: Took me a while to think about it but I think you're right. I updated the code. Thanks!

**Review by kinnison — APPROVED** [review]
Looks good to me.  I've not confirmed behaviour because I'm in the middle of a different issue, but it seems sensible by inspection.

**Comment by GuillaumeGomez** [comment]
Then let's wait for your confirmation. :)

**Comment by Dylan-DPC** [comment]
looks good to me, ping me when ready to merge

**Comment by kinnison** [comment]
Can confirm functioning.
@Dylan-DPC r=me

**Pull request merged**